src/packages/builder/src/codegen/index.ts

```diff
@@ -56,9 +52,22 @@ export const codeGenerator = async () => {
 			},
 		});
 
-		await Promise.all(
-			files.map(async (file) => fs.promises.writeFile(file.filename, file.content, 'utf8'))
-		);
+		// ensure that we have a graphweaver directory
+		const dirPath = path.join(process.cwd(), outdir || './.graphweaver');
+		if (!fs.existsSync(dirPath)) {
+			fs.mkdirSync(dirPath, { recursive: true });
+		}
+
+		const writeOperations = files.flatMap((file) => [
+			fs.promises.writeFile(path.join(process.cwd(), file.filename), file.content, 'utf8'),
+			// We save the types to two locations src and .graphweaver / outdir
```

Comment:
Why do we save the types to two locations?

Reply:
The first location is used by the generated Apollo hooks and the types file is referenced from the other generated files.

The second location is for use outside of the app.

For example, you can specify the location to be inside another app you are working on. Im going to create a docs page on how to do this.